src/main/java/org/prebid/server/bidder/telaria/TelariaBidder.java

```diff
@@ -169,29 +167,36 @@ private static MultiMap headers(BidRequest bidRequest) {
     public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
         try {
             final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
-            return Result.withValues(extractBids(bidResponse));
+            return Result.withValues(extractBids(bidResponse, bidRequest));
         } catch (PreBidException | DecodeException e) {
             return Result.withError(BidderError.badServerResponse(e.getMessage()));
         }
     }
 
-    private List<BidderBid> extractBids(BidResponse bidResponse) {
+    private List<BidderBid> extractBids(BidResponse bidResponse, BidRequest bidRequest) {
         if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
             return Collections.emptyList();
         }
-        return bidsFromResponse(bidResponse);
+        return bidsFromResponse(bidResponse, bidRequest);
     }
 
-    private static List<BidderBid> bidsFromResponse(BidResponse bidResponse) {
+    private static List<BidderBid> bidsFromResponse(BidResponse bidResponse, BidRequest bidRequest) {
         final SeatBid firstSeatBid = bidResponse.getSeatbid().get(0);
         final List<Bid> bids = firstSeatBid.getBid();
+        final List<BidderBid> bidderBids = new ArrayList<>();
+        final List<Imp> imps = bidRequest.getImp();
 
-        if (CollectionUtils.isEmpty(bids)) {
+        if (CollectionUtils.isEmpty(bids) || CollectionUtils.isEmpty(imps)) {
             return Collections.emptyList();
         }
```

Comment:
1. imps cannot be empty, request validator checks it on earlier stage.
2. you can omit whole `if` statement with `CollectionUtils.emptyIfNull(bids).stream()`. Not sure if this will look more readable, check it out.